Tests.Unit.Core.ApplicationServices/ApplicationServices/SystemUsage/ItSystemUsageServiceTest.cs

```diff
@@ -869,6 +866,81 @@ public void RemoveRelevantUnit_Returns_Forbidden()
             Test_Command_Which_Fails_With_Usage_Insufficient_WriteAccess(id => _sut.RemoveRelevantUnit(id, A<Guid>()));
         }
 
+        [Fact]
+        public void Can_AddPersonalDataOption()
+        {
+            //Arrange
+            var itSystem = CreateItSystem();
+            var itSystemUsage = CreateSystemUsage(A<int>(), itSystem);
+
+            itSystemUsage.SensitiveDataLevels = new List<ItSystemUsageSensitiveDataLevel> { new(){ItSystemUsage = itSystemUsage, SensitivityDataLevel = SensitiveDataLevel.PERSONALDATA} };
+
+            var option = A<GDPRPersonalDataOption>();
+
+            ExpectAllowModifyReturns(itSystemUsage, true);
+            _usageRepository.Setup(x => x.GetByKey(itSystemUsage.Id)).Returns(itSystemUsage);
+            var transaction = new Mock<IDatabaseTransaction>();
+            _transactionManager.Setup(x => x.Begin()).Returns(transaction.Object);
+
+            //Act
+            var error = _sut.AddPersonalDataOption(itSystemUsage.Id, option);
+
+            //Assert
+            Assert.True(error.IsNone);
+            _usageRepository.Verify(x => x.Update(itSystemUsage));
+            transaction.Verify(x => x.Commit());
+        }
+
+        [Fact]
+        public void AddPersonalDataOption_Returns_NotFound()
+        {
+            Test_Command_Which_Fails_With_Usage_NotFound(id => _sut.AddPersonalDataOption(id, A<GDPRPersonalDataOption>()));
+        }
+
+        [Fact]
+        public void AddPersonalDataOption_Returns_Forbidden()
+        {
+            Test_Command_Which_Fails_With_Usage_Insufficient_WriteAccess(id => _sut.AddPersonalDataOption(id, A<GDPRPersonalDataOption>()));
+        }
+
+        [Fact]
+        public void Can_RemovePersonalDataOption()
+        {
+            //Arrange
+            var itSystem = CreateItSystem();
+            var itSystemUsage = CreateSystemUsage(A<int>(), itSystem);
+
+            itSystemUsage.SensitiveDataLevels = new List<ItSystemUsageSensitiveDataLevel> { new(){ItSystemUsage = itSystemUsage, SensitivityDataLevel = SensitiveDataLevel.PERSONALDATA} };
+
+            var option = A<GDPRPersonalDataOption>();
+            itSystemUsage.PersonalDataOptions = new List<ItSystemUsagePersonalData> { new(){ItSystemUsage = itSystemUsage, PersonalData= option} };
+            
+            ExpectAllowModifyReturns(itSystemUsage, true);
+            _usageRepository.Setup(x => x.GetByKey(itSystemUsage.Id)).Returns(itSystemUsage);
+            var transaction = new Mock<IDatabaseTransaction>();
+            _transactionManager.Setup(x => x.Begin()).Returns(transaction.Object);
+
+            //Act
+            var error = _sut.RemovePersonalDataOption(itSystemUsage.Id, option);
+
+            //Assert
+            Assert.True(error.IsNone);
+            _usageRepository.Verify(x => x.Update(itSystemUsage));
+            transaction.Verify(x => x.Commit());
+        }
+
+        [Fact]
+        public void RemovePersonalDataOption_Returns_NotFound()
```

Comment:
missing some code branches here such as validation of access rights etc.. goes for both methods